asan.mixin

```diff
@@ -2,8 +2,8 @@
     "build": {
         "asan-gcc": {
```

Comment:
Are you using this mixin with `clang`? If yes, the name of the mixin seems misleading and should maybe be changed.

Reply:
The sanitizer jobs we are currently enabling on ROS2 CI will be on gcc. Will change the name to a more generic "asan" as these flags worked with clang as well during my local testing

Reply:
Is it fine to change the name here now though? This is an already existing mixin and changing the name will break if being used somewhere.

Reply:
That is ok with me. When users update the mixin repo they will get an error message that the old mixin name isn't available anymore and discovering the new name should be straight forward.

Reply:
Updated the name